src/components/Form.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+
+function Form() {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const titleChangeHandler = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const authorChangeHandler = (e) => {
+    setAuthor(e.target.value);
+  };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+
+    // submit data
+    // if (!title.trim() || !author.trim()) return;
+  };
+  return (
+    <div className="form-wrapper row">
+      <form onSubmit={submitHandler}>
+        <input
+          type="text"
+          name="title"
+          value={title}
+          aria-label="Book title input"
+          onChange={titleChangeHandler}
+        />
+        <input
+          type="text"
+          name="author"
+          value={author}
+          aria-label="Book author input"
+          onChange={authorChangeHandler}
+        />
+        <button type="submit">Submit</button>
+      </form>
+    </div>
+  );
+}
+
+export default Form;
```

Comment:
- [ ] Kindly make sure that when this form is submitted an action is dispatched and the book is saved in the redux store. like this example below 👇 
```.jsx
// Note that this is just an example you can address this as you want but make sure that book is saved in the
// redux store using `react-redux`(useDispatch)



import React from "react"
import { useDispatch } from "react-redux"
import { addBook } from "path/to/booksSlice"

function NewBookForm() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
  const books = useSelector((state) => state.books.books)
  const dispatch = useDispatch()

  // Other logic that controls form state

  return (
    <div>
      {/* Rest of form JSX */}
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
    </div>
  )
}

export default NewBookForm
```

Reply:
Fixed.